src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,46 +1,97 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo,
+  setSelectedTodo: (todo: Todo | null) => void,
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  const loadUser = async () => {
+    setHasError(false);
```

Comment:
Use `setIsLoading(true);` before starting to load the user data in `loadUser` function to show the loader while waiting for the data.